### PR TITLE
Remove/Deprecate HLC handling code

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
- Removed classes:
  - HLCSegmentName
  - SegmentName.RealtimeSegmentType
  - PinotRealtimeSegmentManager
  - ReplicationUtils
  - StreamConfig.ConsumerType
- Deprecated classes:
  - SegmentName
  - PartitionLevelStreamConfig

### CODE REVIEW

**Comment by xiangfu0** [comment]
I think we also want to deprecate `_replicasPerPartition` in `SegmentsValidationAndRetentionConfig`?

**Review by xiangfu0 — APPROVED** [review]

**Review by walterddr — COMMENTED** [review]
missed this PR but this is AWESOME!!!!

File: pinot-common/src/main/java/org/apache/pinot/common/utils/SegmentName.java [hunk 1]
```diff
@@ -18,80 +18,16 @@
...
-
-  public String getPartitionRange() {
-    throw new RuntimeException("No partitionRange in " + getSegmentName());
-  }
-
-  public static boolean isHighLevelConsumerSegmentName(String segmentName) {
-    int numSeparators = getNumSeparators(segmentName);
-    return numSeparators == 2 || numSeparators == 4;
+@Deprecated
+public class SegmentName {
```
Comment (walterddr) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/SegmentName.java:22]: this class can be removed?

Comment (Jackie-Jiang) [reply pinot-common/src/main/java/org/apache/pinot/common/utils/SegmentName.java:22]: Kept `SegmentName` and `PartitionLevelStreamConfig` for compatibility. Will remove in next release

**Pull request merged**